### PR TITLE
API-27979-claims-v1-index-bgs-diff for BGS flipper

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

With claims_status_v1_bgs_enabled set to 'On' in staging, we noticed some differences in the data returned for claim status index and this addresses those.
- requested_decision should return the same data when claims_status_v1_bgs_enabled feature is on
- waiver_submitted should return the same data when claims_status_v1_bgs_enabled feature is on
 

## Related issue(s)
- [API-27979](https://jira.devops.va.gov/browse/API-27979)

## Testing done

- Postman

## What areas of the site does it impact?
[*(Describe what parts of the site are impacted and*if*code touched other areas)*](modified:   modules/claims_api/lib/claims_api/evss_bgs_mapper.rb)

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/lib/claims_api/evss_bgs_mapper.rb [hunk 1]
```diff
@@ -25,7 +25,10 @@ def map_and_build_object # rubocop:disable Metrics/MethodLength
       claim['data']['development_letter_sent'] = @data['development_letter_sent']
       claim['data']['decision_letter_sent'] = @data['decision_notification_sent']
       claim['data']['documents_needed'] = @data['wwsnfy'].present?
+
       claim['data']['waiver5103_submitted'] = @data['filed5103_waiver_ind'] == 'Y'
+      claim['data']['decision_requested'] = @data['filed5103_waiver_ind'] == 'Y'
```
Comment (FonzMP) [inline modules/claims_api/lib/claims_api/evss_bgs_mapper.rb:30]: interesting.

Comment (FonzMP) [reply modules/claims_api/lib/claims_api/evss_bgs_mapper.rb:30]: why do we have 2 values set by the same thing.. that seems odd, but 🤷

**Review by FonzMP — APPROVED** [review]
:shipit:

File: modules/claims_api/lib/claims_api/evss_bgs_mapper.rb (hunk 1 above)
Comment (stiehlrod) [reply modules/claims_api/lib/claims_api/evss_bgs_mapper.rb:30]: I checked with Kayla, and that is part of the fun of v1

**Pull request merged**